text/0039-service-overhaul.md

```diff
@@ -0,0 +1,462 @@
+                ],
+            }
+        }
+    ]
+}
+```
+
+- `"format_version"`: the format version, changed when the JSON is no longer backward compatible.
+- `"services"`: array of services.
+  - `"name"`: name showed in OBS.
+  - `"common"` (optional): is the service only shown when the user has clicked on `Show all`.
+  - `"more_info_link"` (optional): link with more info about the service.
+  - `"stream_key_link"` (optional): link where the user can find his stream key.
+  - `"alt_names"` (optional): alternative service name.
+  - `"servers"`: array of servers.
+    - `"name"`: name of the server.
+    - `"url"`: url of the server.
+  - `"recommended"` (it depends): "recommended" settings that become the default when using this service.
+    - `"output"` (not optional if using a protocol different from RTMP(S)): indicates which output **shall** be used because the service use a protocol different from RTMP(S).
+    - `"keyint"` (optional): default keyframe interval for this service.
+    - `"bframes"` (optional): default b-frames for this service.
+    - `"profile"` (optional): default H264 encoder profile for this service.
+    - `"x264opts"` (optional): default option for x264 for this service.
+    - `"max video bitrate"` (optional): maximum video bitrate for this service.
+    - `"max audio bitrate"` (optional): maximum audio bitrate for this service.
+    - `"max fps"` (optional): maximum frame per second for this service.
+    - `"supported resolutions"` (optional): array of supported resolutions for this service.
+    - `"bitrate matrix"` (optional): array of maximum bitrate for a certain resolution/FPS combo for this service.
+      - `"res"`: resolution.
+      - `"fps"`: frame per seconds.
+      - `"max bitrate"`: maximum video bitrate.
+
+##### Issues with this format
+- The naming scheme is a mix of object name with space and some with underscores.
+- About `"output"`, OBS consider every service as RTMP if not added and it's not a recommended settings at all. It makes OBS use the right protocol for the service. Also this prevent a service of being multi-protocol.
+- About `"recommended"`, most of the options seems to H264 related
+- `"common"`, the name makes it not understandable at the first sight maybe adding some documention would be good thing.
+
+#### New format (WIP)
+The new format will be parsable with a new interface library named `service-json-parser` to allow first-party plugins to read this format rather than recreating the wheel.
+
+The new format will be considered as the version 4.
+
+This format allow some specific additions parsed by getter added to the concerned plugin.
+
+Multi-service plugins should have a JSON root that look like this:
+```json
+{
+    "format_version": 4,
+    "services": []
+}
+```
+`"services"` is an array of service objects.
+
+Service plugin like integrations one should have a JSON root that look like this:
+```json
+{
+    "format_version": 4,
+    "service": {}
```

Comment:
This feels like unnecessary semantics, as an array can contain a single entry. For consistency and ease of use, I'd prefer just the plural.